src/main/kotlin/org/jetbrains/teamcity/rest/api.kt

```diff
@@ -32,6 +32,8 @@ interface VcsRootLocator {
 interface BuildLocator {
     fun fromConfiguration(buildConfigurationId: BuildConfigurationId): BuildLocator
 
+    fun buildNumber(buildNumber: String): BuildLocator
```

Comment:
Why name of this method is different from names of other methods in this class which do similar things (filter results by some parameter)? It makes it harder to understand what is it supposed to do. If you really want to add this method to BuildLocator interface it would be better to name it 'withBuildNumber' or even 'withNumber' because all this class is about builds.

But I'm not sure that we really need filtering by build number here. Is there any real cases when there are several builds with the same number? If not all calls of this method would look like `builds().fromConfiguration(xxx).withNumber(yyy).latest()` and in that case it would be better to introduce a special method into TeamCityInstance instead (`build(configuration: BuildConfigurationId, buildNumber: String)`).